app/assets/javascripts/discourse/app/initializers/url-redirects.js

```diff
@@ -9,8 +9,9 @@ export default {
     const currentUser = container.lookup("current-user:main");
     if (currentUser) {
       const username = currentUser.get("username");
+      const escapedUsername = username.replace(/\./g, "\\.");
```

Comment:
is $ somehow allowable? is ? or + somehow allowable? 

`replace(/[.*+?^${}()|[\]\\]/g, '\\$&');` 

https://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex

for extra safety?

Reply:
I don't think it's possible to allow dollar sign or plus per https://meta.discourse.org/t/unicode-usernames-and-group-names/117737, but I guess extra safety wouldn't hurt in case we allow these special characters in the future 😄